zcash_primitives/src/note_encryption.rs

```diff
@@ -483,15 +483,15 @@ pub fn try_sapling_compact_note_decryption<P: consensus::Parameters>(
 
 /// Recovery of the full note plaintext by the sender.
 ///
-/// Attempts to decrypt and validate the given `enc_ciphertext` using the given `ovk`.
+/// Attempts to decrypt and validate the given `enc_ciphertext` using the given `ock`.
 /// If successful, the corresponding Sapling note and memo are returned, along with the
 /// `PaymentAddress` to which the note was sent.
 ///
-/// Implements section 4.17.3 of the Zcash Protocol Specification.
-pub fn try_sapling_output_recovery<P: consensus::Parameters>(
+/// Implements part of section 4.17.3 of the Zcash Protocol Specification.
+/// for decryption using a Full Viewing Key see [try_sapling_output_recovery].
+pub fn try_sapling_output_recovery_with_ock<P: consensus::Parameters>(
     height: u32,
-    ovk: &OutgoingViewingKey,
-    cv: &edwards::Point<Bls12, Unknown>,
+    ock: &[u8],
```

Comment:
```suggestion
    ock: &[u8; 32],
```

Reply:
Hmm, this is good for using with a standalone `ock`, but awkward to use with a just-derived `ock`. `prf_ock(&ovk, &cv, &cmu, &epk)` returns `blake2b_simd::Hash`, which offers the following APIs:
- `Hash::as_array() -> &[u8; 64]` - panics in debug mode if the output length is not 64, and it isn't for `prf_ock` (which uses `BLAKE2b-256`).
- `Hash::as_bytes() -> &[u8]` - this suggestion would require changing the existing `prf_ock(&ovk, &cv, &cmu, &epk).as_bytes()` call to `&prf_ock(&ovk, &cv, &cmu, &epk).as_bytes().try_into().unwrap()`

Reply:
OK, never mind. We can clean this up later.

Reply:
I think the cleanup should probably be to have `prf_ock(...)` return `&[u8; 32]`, btw.